pkg/controller/jobs/pod/pod_controller_test.go

```diff
@@ -489,16 +489,17 @@ type podErrFinalizeMock struct {
 }
 
 var (
-	errMock = errors.New("mock client error")
-	counter = 0
+	errMock            = errors.New("mock client error")
+	shouldFailFinalize = true
```

Comment:
This won't prevent a data race.
You have two alternatives: use the `atomic` package.

Or implement the failure in the client, instead of introducing a PodMock

https://github.com/kubernetes/kubernetes/blob/6f5fa2eb2f4dc731243b00f7e781e95589b5621f/pkg/scheduler/util/utils_test.go#L279

Reply:
I don't think that race could happen here. The reason for that error, is that -count=3 will execute each test 3 times. So I need to set shouldFailFinalize back to `true` in the mock. The tests won't run in parallel unless we're calling `t.Parallel()`as far as I remember.